src/_pytest/fixtures.py

```diff
@@ -1529,34 +1530,56 @@ def sort_by_scope(arg_name: str) -> int:
         return initialnames, fixturenames_closure, arg2fixturedefs
 
     def pytest_generate_tests(self, metafunc: "Metafunc") -> None:
+        """Generate new tests based on parametrized fixtures used by the given metafunc"""
+
+        def get_parametrize_mark_argnames(mark: Mark) -> Sequence[str]:
+            if "argnames" in mark.kwargs:
+                argnames = mark.kwargs[
+                    "argnames"
+                ]  # type: Union[str, Tuple[str, ...], List[str]]
+            else:
+                argnames = mark.args[0]
+            if not isinstance(argnames, (tuple, list)):
+                argnames = [x.strip() for x in argnames.split(",") if x.strip()]
+            return argnames
+
         for argname in metafunc.fixturenames:
-            faclist = metafunc._arg2fixturedefs.get(argname)
-            if faclist:
-                fixturedef = faclist[-1]
+            # Get the FixtureDefs for the argname.
+            fixture_defs = metafunc._arg2fixturedefs.get(argname)
+            if not fixture_defs:
+                # Will raise FixtureLookupError at setup time if not parametrized somewhere
+                # else (e.g @pytest.mark.parametrize)
+                continue
+
+            # The test itself parametrizes using this argname, give it
+            # precedence.
+            if any(
+                argname in get_parametrize_mark_argnames(mark)
+                for mark in metafunc.definition.iter_markers("parametrize")
+            ):
+                continue
```

Comment:
Noticed this should be `continue` instead of `break`, otherwise we would skip evaluating the rest of the argnames for possible fixture parametrization (`test_override_parametrize_fixture_and_indirect` demonstrates this).